packages/video_player/video_player/lib/video_player.dart

```diff
@@ -545,6 +547,22 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     await _videoPlayerPlatform.setLooping(_textureId, value.isLooping);
   }
 
+  /// Returns if caching is supported for network media.
+  Future<bool> isCacheSupportedForNetworkMedia(String uri) async {
+    if (_isDisposedOrNotInitialized) {
+      return false;
+    }
+    return _videoPlayerPlatform.isCacheSupportedForNetworkMedia(uri);
+  }
+
+  /// Clears the cache of the player, returns true if succeeded.
+  Future<bool> clearCache() async {
```

Comment:
@TimHoogstrate Ping on this question, since it's fundamental to the high-level design.

Reply:
I agree, I think having a cache configuration per network call can be a bit confusing and can lead to unexpected behaviors if the caller changes the option set for different code path in there code.

One approach is to let the caller pass in the cache store per video instead of the config, such as the below example. Passing the cache store would indicate that this video should be cached, if null then no cache.

```dart
VideoPlayerController controller = VideoPlayerController.networkUrl(
        uri,
        videoPlayerOptions: VideoPlayerOptions(
            cacheStore: networkVideoCacheStore),
      );
```

Then to make it easy on consumers a DefaultCacheStore can be provided where users can pass that in; however, if they want to further customize the caching/partitioning of videos being streamed they have the full ability to implement the interface in a different way as long as they abide by the interface contract/exception pattern.

@stuartmorgan/ @TimHoogstrate what are your thoughts on that approach?

Reply:
One other thought I had on this, is the cached network store can be implemented as a parallel class where the VideoPlayerController exposes a streamed operation and the wrapper class just consumes that stream and caches it while also passing it to the necessary render widget without having to do a double read of the stream info.